pandas/io/parsers.py

```diff
@@ -889,7 +933,16 @@ def _get_options_with_defaults(self, engine):
 
         for argname, default in _parser_defaults.items():
             value = kwds.get(argname, default)
-
+            if argname in _pyarrow_unsupported:
+                if engine == "pyarrow" and value != default:
+                    raise ValueError(
+                        f"The {repr(argname)} option is not supported with the "
+                        f"'pyarrow' engine"
+                    )
+            if argname == "iterator" and engine == "pyarrow":
```

Comment:
you have *some* options covered above on L437 (iterator and chunksize not none), ideally prefer them *all* to be caught here